src/rpc/request.cpp

```diff
@@ -173,6 +173,18 @@ void JSONRPCRequest::parse(const UniValue& valRequest)
     // Parse id now so errors from here on will have the id
     id = request.find_value("id");
 
+    // Check for JSON-RPC 2.0 (default 1.1)
+    m_json_version = JSONVersion::JSON_1_BTC;
+    const UniValue& valJsonRPC = request.find_value("jsonrpc");
+    if (!valJsonRPC.isNull()) {
+        if (!valJsonRPC.isStr()) {
+            throw JSONRPCError(RPC_INVALID_REQUEST, "jsonrpc field must be a string");
+        }
+        if (valJsonRPC.get_str() == "2.0") {
```

Comment:
In commit "rpc: identify JSON-RPC 2.0 requests" (9e1c6fa84eef606d5504ebcbbd2a2ca7a4f748dc)

The spec says this field 'MUST be exactly "2.0"' (https://www.jsonrpc.org/specification#request_object), so maybe it should be an error if the field is set and contains any other value. This way if there is ever a JSON-RPC 2.1 or 3.0 protocol, clients can easily check for support without getting back a strange JSON 1.1 response. FWIW, chatgpt suggestd the following response:

```json
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32600,
        "message": "Invalid Request",
        "data": "JSON-RPC version not supported."
    },
    "id": null
}
```

And setting the id field to the actual request ID if possible to determine